src/DigitalSubscriberAppBanner/index.tsx

```diff
@@ -1,39 +1,38 @@
 import React from 'react';
 import { AppBanner } from '../AppBanner';
-import type { OphanComponentEvent } from '@guardian/libs';
-import type { BrazeClickHandler } from '../utils/tracking';
+import type { TrackClick } from '../utils/tracking';
 import { COMPONENT_NAME } from './canRender';
 export { COMPONENT_NAME };
 
 export type BrazeMessageProps = {
+    ophanComponentId?: string;
     header?: string;
     body?: string;
+    cta?: string;
+    imageUrl?: string;
```

Comment:
I think it's confusing to add these as props here as they aren't supported by this component? (i.e. the image and cta are hard coded in this case).

Reply:
It raises the question: why do we have/support the `DigitalSubscriberAppBanner` template? The only difference between this and `AppBanner` is that the CTA and URL are hardcoded which, moving to a greater self-serve world, is something we do not want to encourage?

Reply:
In the Accessibility PR, which builds on this PR, I've factored out the local `BrazeMessageProps` type from each banner `index.tsx` to an all-encompassing imported type (maybe it should be an `interface`?) - which you can see defined here: https://github.com/guardian/braze-components/pull/400/files#diff-3341aff8c9584e4e301a155c9d5b5f58b589524f09dd49b5743eef8515798994 ... but I'm happy to take out unnecessary items from the local declarations in this PR.

Reply:
Done

Reply:
> In the Accessibility PR, which builds on this PR, I've factored out the local BrazeMessageProps type from each banner index.tsx to an all-encompassing imported type (maybe it should be an interface?)

Hmm wonder if that's what we want. I'll use epics here as an example but the same is true for banners. High level, there is a common interface which all Braze epics have which is defined here: https://github.com/guardian/braze-components/blob/main/src/BrazeEndOfArticleComponent.ts#LL49-L56C3 So that's everything the platform will provide to the high level `BrazeEndOfArticleComponent` component. At this level the `brazeMessageProps` are very generic and represent the data we get from Braze. Then downstream of that each individual component specifies narrower types, which outline the props that the specific component wants. So taking the `BrazeDownToEarthNewsletterEpic` as an example, it's props are defined here: https://github.com/guardian/braze-components/blob/main/src/DownToEarthNewsletterEpic/index.tsx#L19, so this narrows down and makes more specific the higher level props. Mixing all possible props that each component might want in a higher level type/interface at the top feels like it might be confusing as it'll be harder to know what props/values from Braze an individual component cares about. What do you think? (Happy to pair on any of this if it's useful or if I'm not making sense!)

Reply:
Hey yes I think moving to more generic/configurable components on the whole is a good strategy. The approach we've used in the past on this project is create more bespoke/specific templates and then if they prove useful to create something more generic (which is how `DigitalSubscriberAppBanner` -> `AppBanner` happened). Great question about `DigitalSubscriberAppBanner`, I'd be surprised if that is still in use anywhere. If we can get our marketing colleagues to confirm it's unused we could remove it (I think that probably belongs in its own PR).